worker/src/core/loaders/message-worker/index.ts

```diff
@@ -221,6 +221,21 @@ const enqueueAndSend = async (): Promise<void> => {
             jobId,
           })
         }
+        const whileLoopTimeTakenMs = Date.now() - start
```

Comment:
If I'm not mistaken, we're interested in how long the entire while loop takes (since that's the actual rate of sending) and the `waitForMs(wait)` is to make sure that the while loop takes at least 1 second? So if we move this up, that might mess up our monitoring?

(This discussion would be superseded by @stanleynguyen's suggestion that we should be monitoring the send rate of **each** message; see the Slack thread for the discussion hmm)